go/ec2/RegionsAndZones/RegionsAndZones.go

```diff
@@ -0,0 +1,82 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT-0
+// snippet-start:[ec2.go.regions_and_zones]
+package main
+
+// snippet-start:[ec2.go.regions_and_zones.imports]
+import (
+    "fmt"
+
+    "github.com/aws/aws-sdk-go/aws/session"
+    "github.com/aws/aws-sdk-go/service/ec2"
+)
+// snippet-end:[ec2.go.regions_and_zones.imports]
+
+// GetRegions retrieves the AWS Regions where Amazon Elastic Compute Cloud (Amazon EC2) is available.
+// Inputs:
+//     sess is the current session, which provides configuration for the SDK's service clients
+// Output:
+//     If success, the list of AWS Regions and nil
+//     Otherwise, nil and an error from the call to DescribeRegions
+func GetRegions(sess *session.Session) (*ec2.DescribeRegionsOutput, error) {
+    // snippet-start:[ec2.go.regions_and_zones.regions]
+    svc := ec2.New(sess)
+
+    resultRegions, err := svc.DescribeRegions(nil)
+    // snippet-end:[ec2.go.regions_and_zones.regions]
+    if err != nil {
+        return nil, err
+    }
+
+    return resultRegions, nil
+}
+
+// GetZones retrieves the availability zones within the current region.
```

Comment:
**Availability Zones** within the current **AWS Region**

Reply:
Fixed